cli/flox-rust-sdk/src/models/manifest/composite/mod.rs

```diff
@@ -60,21 +65,24 @@ impl<Key: Into<String>> FromIterator<Key> for KeyPath {
 ///
 /// Warnings are not errors, but they may indicate
 /// that the user should review the merged manifest or its dependencies.
-///
-/// Currently, the only warning is that a value is being overridden,
-/// but more warnings may be added in the future.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
 #[must_use]
 pub enum Warning {
     Overriding(KeyPath),
+    /// Currently, the only warning is that a value is being overridden,
+    /// but more warnings may be added in the future. This placeholer prevents
+    /// linting from complaining about irrefutable matches and let statements.
+    Placeholder(),
 }
```

Comment:
I think [`non_exhaustive`](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute) is the language construct that should avoid that too, did you have a look at that?